### PR TITLE
feat: add a multi-precision module for testing

### PR DESCRIPTION
**Pull request by lgrcia**
This adds a multi-precision module to compute all starry quantities using the mpmath package with arbitrary precision.

### CODE REVIEW

**Comment by soichiro-hattori** [comment]
@lgrcia: I'm happy for this to be merged! 
Will these module need to be documented or is it only going to be used for testing?

**Comment by lgrcia** [comment]
I think mainly testing. I also had in mind a CI (maybe not on GitHub) to monitor the precision after each PR. We can talk about it in #207.

**Comment by lgrcia** [comment]
Before I merge, @dfm, do you think this is a meaningful thing to push to main? It's kind of a big addition only for testing so I'd love to have your feedback.

File: tests/precision/test_flux.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        ha="left",
+        va="top",
+        alpha=0.75,
+    )
+    plt.tight_layout()
+    plt.savefig("flux_precision.png")
+
+
+if __name__ == "__main__":
+    plot_flux_precision()
```
Comment (dfm) [inline tests/precision/test_flux.py:138]: I'm not 100% sure I'd put these plotting functions in `tests`. Maybe `examples` or `scripts` or something? Or maybe at least split all the plotting tests out into a separate script?

**Review by dfm — APPROVED** [review]
LGTM! Tiny comment inline, but this makes sense to me!

**Pull request merged**